### PR TITLE
Env Varibles for all-the-things

### PR DESCRIPTION
**Pull request by eladg**

### CODE REVIEW

File: .env-example [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Example ENV VARIABLES
-REDIS_URL=redis://redis:623e9c0cb0a185b9cac9b3c8geal3812@joe.redis.com:12345/
-ACCESS_CONTROL_ALLOW_ORIGIN=*
-AWS_ACCESS_KEY_ID=ABCDEFGHIJKLMNOPQRST
-AWS_SECRET_ACCESS_KEY=A1b2C3D4/e5-f6g7j2K2x2i13n1+0x21xp7Ux-sl
\ No newline at end of file
+REDIS_URL="redis://redis:623e9c0cb0a185b9cac9b3c8geal3812@joe.redis.com:12345/"
+AWS_ACCESS_KEY_ID="ABCDEFGHIJKLMNOPQRST"
```
Comment (gaaady) [inline .env-example]: we should change the AWS access key.
there are spam bots crawling Github for secret keys and then using them to allocate EC2 instances and sending spam
.env-example should not contain secret

File: midburn_queue.rb [hunk 2]
```diff
@@ -21,40 +28,48 @@ class MidburnQueue < Sinatra::Base
...
+  def set_response_headers
+    response.headers["Access-Control-Allow-Headers"] = "X-Requested-With, X-HTTP-Method-Override, Content-Type, Cache-Control, Accept"
+    response.headers["Access-Control-Allow-Origin"] = ACCESS_CONTROL_ALLOW_ORIGIN
+  end
+
+  def load_params
+    return @params if @request_body_read
     begin
-      JSON.parse(request.body.read)
+      @params = JSON.parse(request.body.read)
```
Comment (gaaady) [inline midburn_queue.rb]: which JSON parser are we using?
consider using https://github.com/ohler55/oj

File: midburn_queue.rb [hunk 3]
```diff
@@ -21,40 +28,48 @@ class MidburnQueue < Sinatra::Base
...
+      @request_body_read = true
     rescue Exception => e
-      {}
+      @params = {}
```
Comment (gaaady) [inline midburn_queue.rb]: small issue @request_body_read isn't set to true in this case

File: .env-example (hunk 1 above)
Comment (eladg) [inline .env-example]: `ABCDEFGHIJKLMNOPQRST` is not a secret, it's based on a very popular song https://www.youtube.com/watch?v=75p-N9YKqNo ;)

File: midburn_queue.rb (hunk 2 above)
Comment (eladg) [inline midburn_queue.rb]: Good point, I like OJ, but for such a small payload (i.e. `{"email":"user@example.com"}`), I don't think we will gain a lot by switching.

File: .env-example (hunk 1 above)
Comment (gaaady) [inline .env-example]: :)
is it the same case for the redis usernaame & password? and the keys in the history?

Comment (eladg) [inline .env-example]: those are made up. joe.redis.com is not even a valid DNS namespace.

File: midburn_queue.rb (hunk 3 above)
Comment (eladg) [inline midburn_queue.rb]: yes, indeed. this should be fixed.

Comment (eladg) [inline midburn_queue.rb]: testing slack integration, please ignore this.

Comment (eladg) [inline midburn_queue.rb]: maybe now??

File: midburn_queue.rb (hunk 2 above)
Comment (eladg) [inline midburn_queue.rb]: looked into it. We are using `multi_json` which is a Sinatra and Resque dependency (each for their own reasons - read more here: https://github.com/resque/resque/issues/903). I don't see enough value for us to use 2 different libraries for such a small payload. Will bring more troubles then value.

**Pull request merged**